### PR TITLE
Move from `tracing` to `log` + `profiling`

### LINKED ISSUE #1189: Tracing should be optional (especially in dependencies)
**Is your feature request related to a problem? Please describe.**
`tracing` features are enabled unconditionally for some dependencies which can add performance overhead. It's also often used as debug feature and release builds might want to avoid it.

e.g. a crate downstream noticed a lot of overhead when tracing is enabled in gpu-alloc, but wgpu provides no way to opt out of this feature because it's always enabled: https://github.com/gfx-rs/wgpu/blob/8ed543f26a379a2d165f1592710e7f882a15b862/wgpu-core/Cargo.toml#L36-L37

Whether tracing is enabled or disabled can be controlled from the tracing crate's API, but downstream crates not interested in tracing probably shouldn't need to be aware of this.

**Describe the solution you'd like**
`tracing` probably shouldn't be unconditionally enabled. Maybe wgpu could forward the existing `tracing` feature, or possibly not enable it for dependencies at all.

We could also feature-gate tracing inside of wgpu's span macro or other places, e.g. use `#[cfg(feature = "tracing")]` to avoid creating tracing spans in https://github.com/gfx-rs/wgpu/blob/780637eec029097e37dff9238abca7248354b465/wgpu-core/src/lib.rs#L251

**Describe alternatives you've considered**
Downstream crates opting out of tracing by using the API in the `tracing` crate

**Additional context**
It was mentioned on Discord that significant amounts of time are spent in gpu-alloc's tracing and places like https://github.com/gfx-rs/wgpu/blob/f0556d7d8b05295956248d6384cffb76bdb12707/wgpu-core/src/command/render.rs#L2072

We might also want to consider making tracing optional in wgpu-rs for some of the same reasons above.

### LINKED ISSUE #1211: Convert `tracing` to `log` + `profiling`
`tracing` has some performance implications and also mixes profiling spans and with logging. Due to a discussion on the matrix, we've decided to move all spans to the `profiling` crate (with no features enabled) and all logging to the `log` crate.

### PR DESCRIPTION
**Pull request by cwfitzgerald**
**Connections**

Fixes #1211. Fixes #1189.

**Description**

Moves from `tracing` to `log` + `profiling`. I also removed some of the offending spans that were only for pushing commands onto a vector of commands.

**Testing**

Tested on wgpu-rs

### CODE REVIEW

File: Cargo.lock [hunk 1]
```diff
@@ -36,15 +36,6 @@ dependencies = [
  "xml-rs",
 ]
 
-[[package]]
```
Comment (kvark) [inline Cargo.lock:39]: whoa, 190 LOC cleared up from our Cargo lock?
I'm eager to recount our dependencies after this :)

File: wgpu-core/Cargo.toml [hunk 2]
```diff
@@ -27,16 +27,17 @@ arrayvec = "0.5"
 bitflags = "1.0"
 copyless = "0.1"
 fxhash = "0.2"
+log = { version = "0.4" }
```
Comment (kvark) [inline wgpu-core/Cargo.toml:30]: is it intentionally not just `log = "0.4"`?

**Review by kvark — APPROVED** [review]
LGTM, thank you!
please proceed when ready

File: Cargo.lock (hunk 1 above)
Comment (cwfitzgerald) [reply Cargo.lock:39]: 88 build jobs -> 84 building just wgpu-core

File: wgpu-core/Cargo.toml (hunk 2 above)
Comment (cwfitzgerald) [reply wgpu-core/Cargo.toml]: Nah, just expected to need to turn off features, but didn't need to.

**Comment by cwfitzgerald** [comment]
bors r=kvark

Got https://github.com/aclysma/profiling/pull/13 in there too.

**Pull request merged**